apps/web/src/studio/LocalStudioAuthenticator.tsx

```diff
@@ -60,6 +46,34 @@ export function LocalStudioAuthenticator() {
     // Protect against XSS attacks via the javascript: pseudo protocol
     assertProtocol(parsedRedirectURL);
 
+    // Parse the current URL, we will need it later
+    const currentURL = new URL(window.location.href);
+
+    // If the user is not logged in, redirect to the login or signup page
+    if (!currentUser) {
+      // If user is loading, wait for user to be loaded
+      if (!isLoading) {
+        /*
+         * If the user has logged in before, redirect to the login page.
+         * After authentication, redirect back to the this /local-studio/auth path.
+         */
+        if (novuOnboardedCookie.get()) {
+          return redirectToLogin({ redirectURL: window.location.href });
+        }
+
+        /*
+         * If the user hasn't logged in before, redirect to the login page.
+         * After authentication, redirect back to the this /local-studio/auth path and
+         * remember that studio needs to be in onboarding mode.
+         */
+        // currentURL.searchParams.append('studio_path_hint', ROUTES.STUDIO_ONBOARDING);
+
+        return redirectToSignUp({ redirectURL: currentURL.href });
+      }
+
+      return;
+    }
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
☑ **chore:** This `useEffect` is getting pretty huge and hard to follow. Please put a TODO to refactor this to a standalone hook and we can determine how to break it down a bit‏